src/core/reducers/data/navigations.ts

```diff
@@ -26,11 +26,25 @@ export default function updateNavigations(state: State = DEFAULTS, action: Actio
 }
 
 export const updateSearch = (state: State, payload: Actions.Payload.Search) => {
-  if (payload.clear) {
-    const byId = state.allIds.reduce((navs, nav) =>
-      Object.assign(navs, { [nav]: { ...state.byId[nav], selected: [] } }), {});
+  if (payload.clear in state.byId) {
+    const navigationId = <string>payload.clear;
```

Comment:
`payload.clear` will only be handled here if it's an existing navigation id, and thus is a field name
We would suffer here if the field name was `true` or `false`